sdk/core/azure-core/azure/core/pipeline/transport/_base.py

```diff
@@ -703,8 +721,15 @@ def _request(
             request.headers.update(headers)
 
         if content is not None:
+            content_type = request.headers.get("Content-Type")
             if isinstance(content, ET.Element):
                 request.set_xml_body(content)
+            # https://github.com/Azure/azure-sdk-for-python/issues/12137
+            # A string is valid JSON, make the difference between text
+            # and a plain JSON string.
+            # Content-Type is a good indicator of intent from user
+            elif content_type and content_type.startswith("text/"):
```

Comment:
I'm wondering if we've created a bit of a mishmash in how we decide how to serialize/pass request bodies. We are using the `Content-Type` header, which parameters we pass in (`form_data` or `content`/`stream_content`) as well as looking at the content mixed together. For example, we will try to serialize the data as json even if I specify a completely different header (assuming I didn't pass in an `Element` instance that is :)). This is a reasonable first step given where we are.

Reply:
Yes, I'm not really happy by the black magic, it's why I took the simplest approach to avoid two much black magic: when you cannot guess at all if this is a string test, or a string token of JSON.